pkg/operator/connectivitycheckcontroller/connectivity_check_controller.go

```diff
@@ -61,6 +72,17 @@ func (c *connectivityCheckController) WithPodNetworkConnectivityCheckFn(podNetwo
 	return c
 }
 
+// unsupportedConfigOverrides is a partial struct to deserialize just the parts of
+// spec.unsupportedConfigOverrides that we are interested in.
+type unsupportedConfigOverrides struct {
+	Operator struct {
+		DisabledControllers []string `json:"disabledControllers"`
+		OptionalControllers []string `json:"optionalControllers"`
+	} `json:"operator"`
```

Comment:
adding a generic API, but only implementing it here seems wrong. Why not make it specific?

Reply:
Done.